src/EVM/SymExec.hs

```diff
@@ -1117,6 +1119,23 @@ prettyBuf (ConcreteBuf "") = "Empty"
 prettyBuf (ConcreteBuf bs) = formatBinary bs
 prettyBuf b = internalError $ "Unexpected symbolic buffer:\n" <> T.unpack (formatExpr b)
 
+calldataFromCex :: App m => SMTCex -> Expr Buf -> Text -> [AbiType] -> m (Err ByteString)
+calldataFromCex cex buf funName types = do
+  let fullSig = (funName <> "(" <> T.intercalate "," (map abiTypeSolidity types) <> ")")
```

Comment:
We use the function name and types only to construct this signature, Wouldn't it be better to take signature directly?
I was looking at the users of this function, and basically the only user is `symRun`, which constructs the signature anyway.

Reply:
My comment related to this code:

It will be great to have two functions:
- A function to transform a CEX into a high-level solidity call (e.g. f(1,2,3)). 
- A function to transform a high-level solidity call into calldata to run the transaction and verify the result.

So the user will be able to see the values in the call, and then debug an invalid CEX if necessary. Echidna already have the code for these two can that can be easily adapted or even re-used here if necessary.

Reply:
I had a second look and instead of taking `Text` and `[AbyType]`, I think this method should be taking `Sig`.
And we should have a method to convert this to the text representation of a signature. Not sure if `Show` does that for `Sig` or not.
We should not have to repeat code like `(funName <> "(" <> T.intercalate "," (map abiTypeSolidity types) <> ")")`